cmd/datamon/cmd/flags.go

```diff
@@ -432,10 +395,131 @@ func paramsToDestStore(params flagsT,
 	return destStore, nil
 }
 
-func paramsToContributor(_ flagsT) (model.Contributor, error) {
-	return authorizer.Principal(config.Credential)
+/** combined config (file + env var) and parameters (pflags) */
+
+type cliOptionInputs struct {
+	config *CLIConfig
+	params *flagsT
+}
+
+func newCliOptionInputs(config *CLIConfig, params *flagsT) *cliOptionInputs {
+	return &cliOptionInputs{
+		config: config,
+		params: params,
+	}
+}
+
+/** combined config and parameters to internal objects */
+
+func (in *cliOptionInputs) datamonContext(ctx context.Context) (context2.Stores, error) {
```

Comment:
As I mentioned in some other post, we shouldn't be afraid of declaring it a "friend" method like:
```
func (in *cliOptionInputs) DatamonContext(ctx context.Context) (context2.Stores, error)
```
This method is indeed an interface of this object, but for objects in the same package only ("friends")

Reply:
NOTE: this realization came to me upon reading this particular piece of code, it also applies to a lot of things I've done so far...

Reply:
for some reason this reminds me of previous discussions about `let` bindings in Clojure – or more generally lisps with sequential `let`s:

i might write either
```clojure
(let [x (f)
        y (g)]
 (h x y))
```
or
```clojure
(let [x (f)]
  (let [y (g)]
 (h x y)))
```
both are equivalent.

now, i might consider the second snippet somehow suggestive of the fact that the function call `(f)` is conceptually separate from `(g)` in some way.  however, the consensus among seasoned programmers was _not_ to write something that i consider idiomatically suggestive here precisely because it's idiomatic, not necessarily something that programmers will adapt to based on being familiar with the language.

i suppose this "package internal public"/"friend" idiom is similar to what i'd done in private programming wherein i developed some idiom (about `let` bindings, although i've also made up similar conventions like "use single quotes for machine-readable strings and double quotes for human-readable strings" in languages like py and js) that was meaningful to me but not a standard in the ecosystem.  like 90% when i try to bring these conventions into projects, they're not particularly helpful because of the difficulty in having other developers keep track of them.

i'd have no problem if we want to rename some things after the fact, yet "package internal public" is neither a language or ecosystem standard so far as i'm aware, so i suppose attempting to introduce such conventions is going to make things more complex than simpler.

Reply:
see [here](https://github.com/oneconcern/datamon/pull/379#discussion_r374965466) for less rant.

Reply:
ok will probably land a proposal with all this soon